solver-next/edge.go

```diff
@@ -698,6 +749,11 @@ func (e *edge) execOp(ctx context.Context) (interface{}, error) {
 
 	res := results[int(index)]
 
+	for i := range results {
+		if i != int(index) {
+			go results[i].Release(context.TODO())
```

Comment:
https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables ?

Reply:
This doesn't apply here as `i` is evaluated outside the goroutine. If goroutine was wrapped in a closure then we would need to protect it.